src/model/Layer.h

```diff
@@ -14,6 +14,9 @@
 #include "Element.h"
 #include <XournalType.h>
 
+using LayerIndex = std::ptrdiff_t;
```

Comment:
It's not clear whether `LayerIndex` refers to the index of a layer in the list of layers or the index of an element in a layer (I thought it was the former at first reading). Perhaps renaming it as `ElementIndex` and placing it inside of the `Layer` class would be more explicit? Then it can be referred to as `Layer::ElementIndex`, which is unambiguous in the layer context.

Reply:
Excellent! I was debating between `LayerIndex` or `ElementIndex` or `ElementIndexWithinLayer` but they're all messy. I like this solution a lot.